### PR TITLE
Vulkan: Reserve descriptor set 0 for frame-global data, move everything else to set 1

### PR DESCRIPTION
**Pull request by hrydgard**
Make room for frame-global constants (need no rebinding per draw), which will be convenient for stuff like VR and stereo parameters, and possibly some other future uses.

The reason we're not just putting frame global data in set 1 instead and keeping per-draw stuff in set 0, is that when binding new descriptor sets on slots >= 1, slots below are allowed to remain valid as long as the pipeline layouts all point to the same VkDescriptorSetLayout for those lower slots.

In general the recommendation in Vulkan is to keep seldomly changing desc sets in the lower slots and fast-changing ones in higher slots for this reason.

This PR by itself should have no noticable effects on anything.

### CODE REVIEW

File: GPU/Common/PresentationCommon.cpp [hunk 1]
```diff
@@ -233,14 +233,21 @@ bool PresentationCommon::UpdatePostShader() {
...
-		if (!result) {
+		if (stereoShaderInfo) {
+			bool result = CompilePostShader(stereoShaderInfo, &stereoPipeline_);
+			if (!result) {
+				// We won't have a stereo shader. We have to check for this later.
+				delete stereoShaderInfo_;
+				stereoShaderInfo_ = nullptr;
+				stereoPipeline_ = nullptr;
+			} else {
+				stereoShaderInfo_ = new ShaderInfo(*stereoShaderInfo);
```
Comment (unknownbrackets) [inline GPU/Common/PresentationCommon.cpp:244]: Shouldn't we still delete the old one beforehand?  Or if `DestroyStereoShader();` does it, why do we need to delete in these other two paths?

-[Unknown]

File: GPU/Vulkan/VulkanUtil.cpp [hunk 2]
```diff
@@ -85,14 +85,14 @@ void VulkanComputeShaderManager::InitDeviceObjects() {
 
 	std::vector<VkDescriptorPoolSize> dpTypes;
 	dpTypes.resize(2);
-	dpTypes[0].descriptorCount = 8192;
+	dpTypes[0].descriptorCount = 4096;
```
Comment (unknownbrackets) [inline GPU/Vulkan/VulkanUtil.cpp:88]: I feel like we've actually hit these limits though and that's why they're kinda high now?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]
We could probably move a couple things to this, offhand we could move the display rotation I think?

-[Unknown]

File: GPU/Common/PresentationCommon.cpp (hunk 1 above)
Comment (hrydgard) [reply GPU/Common/PresentationCommon.cpp:244]: I dunno, I think I confused myself. I'll clean it up.

File: GPU/Vulkan/VulkanUtil.cpp (hunk 2 above)
Comment (hrydgard) [reply GPU/Vulkan/VulkanUtil.cpp:88]: I guess, they just seemed a bit excessive. But I'll restore for now.

**Comment by hrydgard** [comment]
Yeah rotation would be a good candidate to move here.

**Review by unknownbrackets — APPROVED** [review]

**Pull request merged**